From d61fd80832f558d52de026097d30d695634e8b0f Mon Sep 17 00:00:00 2001
From: Alexander Koskovich <akoskovich@pm.me>
Date: Tue, 06 Oct 2026 14:48:50 -0400
Subject: [PATCH] sepolicy: Allow servicemanager to send callbacks to usbd

usbd gets the USB gadget HAL with waitForService(), which relies on a
callback from servicemanager. Without binder_use() that is denied and
usbd polls instead, adding up to 1s before it configures USB at boot.

Test: Boot, usbd takes 0.165s instead of 1.087s
Change-Id: I72ee3a802dd71134807405a5fc59dacf9a9a5b44
---

diff --git a/private/usbd.te b/private/usbd.te
index c9121a5..6e91333 100644
--- a/private/usbd.te
+++ b/private/usbd.te
@@ -17,4 +17,4 @@
 # Start/stop adbd via ctl.start adbd
 set_prop(usbd, ctl_adbd_prop)
 
-binder_call(usbd, servicemanager)
+binder_use(usbd)
